influxql/iterator.gen.go

```diff
@@ -4558,11 +4558,11 @@ func (itr *stringInterruptIterator) Stats() IteratorStats { return itr.input.Sta
 func (itr *stringInterruptIterator) Close() error         { return itr.input.Close() }
 
 func (itr *stringInterruptIterator) Next() (*StringPoint, error) {
-	// Only check if the channel is closed every 256 points. This
-	// intentionally checks on both 0 and 256 so that if the iterator
+	// Only check if the channel is closed every N points. This
+	// intentionally checks on both 0 and N so that if the iterator
 	// has been interrupted before the first point is emitted it will
 	// not emit any points.
-	if itr.count&0x100 == 0 {
+	if itr.count&5000 == 0 {
```

Comment:
Both the current and previous implementation are checking too frequently. The comparison should not be 0, but rather the same value as the bitmask.

IOW, [x&5000 == 0 is true for 256 values from 0-5000](https://play.golang.org/p/TjMHRXYB8M), but [x&5000 == 5000 is only true once from 0-5000](https://play.golang.org/p/N9Vp_TW7vs).